### PR TITLE
Update: make no-implicit-coercion support autofixing. (fixes #7056)

### LINKED ISSUE #7056: Implement autofix for no-implicit-coercion
**What version of ESLint are you using?**
3.0.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
Seems unnecessary

**What did you do? Please include the actual source code causing the issue.**
I tried to use the --fix command with 

```
"no-implicit-coercion": 2
```

**What did you expect to happen?**
I expected it to fix my source

**What actually happened? Please include the actual, raw output from ESLint.**
It gave me explicit recommendations on what to write in my code, but didn't make the actual changes.

### PR DESCRIPTION
**Pull request by TheSavior**
**What issue does this pull request address?**
https://github.com/eslint/eslint/issues/7056

**What changes did you make? (Give an overview)**
This rule already specifies the fixes to make, this change just makes the code implement those fixes.

**Is there anything you'd like reviewers to focus on?**
Nothing in particular.

### CODE REVIEW

**Comment by TheSavior** [comment]
This eslint bot seems confusing now that commits can be squashed on merge. I added a commit with the correct message format but it still doesn't like it. Does it need to be the first commit?

*Edit* I just squashed my commits to appease it. Seems unnecessary though.

**Comment by platinumazure** [comment]
@TheSavior Yeah, it's talking about the first commit. We should probably remove that check from the bot at some point. Sorry for the inconvenience and thanks for taking the time to squash.

**Comment by platinumazure** [comment]
Unrelated to this PR, I'd like to see the message start with a capital letter. (Don't implement this, I'll create an issue so the team can discuss.)

LGTM, but would like at least one more set of eyes on this. (In addition, this should probably wait at least 2 days for merge, maybe 3 if we factor in the US holiday.)

**Comment by vitorbal** [comment]
LGTM, but the issue still needs to be accepted.

**Comment by kaicataldo** [comment]
Looks like there are some merge conflicts that need to be resolved now (probably from my bug fix PR - sorry!).

**Comment by TheSavior** [comment]
All fixed

File: lib/rules/no-implicit-coercion.js [hunk 1]
```diff
@@ -265,11 +275,11 @@ module.exports = {
                 const operatorAllowed = options.allow.indexOf("+") >= 0;
 
                 if (!operatorAllowed && options.string && isAppendEmptyString(node)) {
-                    context.report(
-                        node,
-                        "use `{{code}} = String({{code}})` instead.", {
-                            code: sourceCode.getText(getNonEmptyOperand(node))
-                        });
+                    const code = sourceCode.getText(getNonEmptyOperand(node, ""));
+
```
Comment (kaicataldo) [inline lib/rules/no-implicit-coercion.js]: `getNonEmptyOperand` only takes one argument (just `node`). Can we get rid of the second argument? Also, mind getting rid of this extra newline while you're at it?

File: lib/rules/no-implicit-coercion.js [hunk 2]
```diff
@@ -241,21 +255,17 @@ module.exports = {
...
 
                 // "" + foo
                 operatorAllowed = options.allow.indexOf("+") >= 0;
                 if (!operatorAllowed && options.string && isConcatWithEmptyString(node)) {
-                    context.report(
-                        node,
-                        "use `String({{code}})` instead.", {
-                            code: sourceCode.getText(getNonEmptyOperand(node))
-                        });
+                    const recommendation = `String(${sourceCode.getText(getNonEmptyOperand(node, ""))})`;
```
Comment (kaicataldo) [inline lib/rules/no-implicit-coercion.js]: Same as line 278 - `getNonEmptyOperand` only takes one argument (just `node`).

**Comment by kaicataldo** [comment]
Just a few small things - thanks for working on this!

**Comment by kaicataldo** [comment]
Sorry! Mouse veered a little too far to the left :(

**Comment by TheSavior** [comment]
Thanks for the comments. All fixed.

**Comment by rpatil26** [comment]
LGTM

**Comment by kaicataldo** [comment]
LGTM. Thanks for contributing to ESLint!

**Pull request merged**